firebase-firestore/src/main/java/com/google/firebase/firestore/remote/FirestoreChannel.java

```diff
@@ -44,6 +42,16 @@
  */
 public class FirestoreChannel {
 
+  public abstract static class Listener<T> {
+    public Listener() {}
+
+    public void onMessage(T message) {}
+
+    public void onClose(Status status) {}
+
+    public boolean callback_fired = false;
```

Comment:
This can be removed (I think), but even if not it should not be part of the Listener interface. You can move it to the callsite if needed.